lib/schema.js

```diff
@@ -579,6 +580,26 @@ Schema.prototype.add = function add(obj, prefix) {
   return this;
 };
 
+Schema.prototype.removeIndex = function removeIndex(index) {
+  if (arguments.length > 1) {
+    throw new Error('removeIndex() takes only 1 argument');
+  }
+
+  if (typeof index !== 'object' && typeof index !== 'string') {
+    throw new Error('removeIndex() may only take either an object or a string as an argument');
+  }
+
+  for (let i = 0; i < this._indexes.length; i++) {
+    if (Object.keys(this._indexes[i][0]) == index || util.isDeepStrictEqual(this._indexes[i][0], index)) {
```

Comment:
`Object.keys(this._indexes[i][0]) == index` -> `this._indexes[i][1].name === index`.

[Indexes have names in MongoDB](https://www.mongodb.com/docs/manual/tutorial/avoid-text-index-name-limit/), so `removeIndex(name)` should remove the index named 'name'.

If the schema index doesn't have a `name` option, then we shouldn't support removing it by `name`. I don't want Mongoose to have to replicate MongoDB's logic for default index names.